files/scripts/asic_status.sh

```diff
@@ -2,8 +2,23 @@
 
 is_chassis_supervisor() {
     if [ -f /etc/sonic/chassisdb.conf ]; then
-        true
-        return
+        # if /etc/sonic/chassisdb.con exists, check platform_env.conf and its supervisor=1 definition
+        # to determinate if it is a Supervisor or Pizzabox with database-chassis support
+        PLATFORM="$(sonic-cfggen -H -v DEVICE_METADATA.localhost.platform)"
+        PLATFORM_ENV_CONF=/usr/share/sonic/device/$PLATFORM/platform_env.conf
+        if [ -f "$PLATFORM_ENV_CONF" ]; then
+            source $PLATFORM_ENV_CONF
+            if [ -v supervisor ]; then
+                if [ $supervisor -eq 1 ]; then
```

Comment:
@mlok-nokia find these checks for supervisor at multiple places. I feel we should just add a flag like may be "diaggregatedT2=1" in one of the platform files (It will be great if there is a way to identify without the need of an explicit flag , eg: https://github.com/sonic-net/sonic-buildimage/blob/master/dockers/docker-fpm-frr/base_image_files/platform_utils). This can be used in bash scripts

we can have a similar API in https://github.com/sonic-net/sonic-buildimage/blob/0955e3edb683a1a655a346d80b15a5924af2543b/src/sonic-py-common/sonic_py_common/device_info.py#L635, like is_diagg_chassis(), and can be used in all python libraries.

This way no need to touch the existing checks for the current supervisor based chassis

Reply:
This modification is because of platform -- mellanox/x86_64-nvidia_sn4280-r0/chassisdb.conf